cognite_toolkit/cognite_modules/examples/cdf_data_pipeline_files_valhall/functions/fu_context_files_oid_fileshare_annotation/pipeline.py

```diff
@@ -0,0 +1,554 @@
+            if split_name[0].isnumeric():
+                names.append(name[len(split_name[0]) + 1 :])
+
+            entities.append(Entity(external_id=asset.external_id, org_name=name, name=name, id=asset.id, type="asset"))
+        except Exception as e:
+            print(
+                f"[ERROR] Not able to get entities for asset name: {name}, id {asset.external_id}. "
+                f"Error: {type(e)}({e})"
+            )
+
+
+def tag_is_dummy(asset):
+    custom_description = (asset.metadata or {}).get("Description", "")
+    return "DUMMY TAG" in custom_description.upper()
+
+
+def process_files(
+    client: CogniteClient,
+    entities: list[Entity],
+    files: dict[str, FileMetadata],
+    annotation_list: dict[Optional[int], list[Optional[int]]],
+    config: AnnotationConfig,
+) -> tuple[int, int]:
+    """Contextualize files by calling the annotation function
+    Then update the metadata for the P&ID input file
+
+    Args:
+        client: client id used to connect to CDF
+        entities: list of input entities that are used to match content in file
+        files: dict of files found based on filter
+        annotation_list: list of existing annotations for input files
+        config: configuration for the annotation process
+
+    Returns:
+        number of annotated files and number of errors
+    """
+    annotated_count = 0
+    error_count = 0
+    annotation_list = annotation_list or {}
+
+    for file_xid, file in files.items():
+        try:
+            # contextualize, create annotation and get list of matched tags
+            entities_name_found, entities_id_found = detect_create_annotation(
+                client, config.match_threshold, file_xid, entities, annotation_list
+            )
+            # create a string of matched tag - to be added to metadata
+            asset_names = shorten(",".join(map(str, entities_name_found)), ASSET_MAX_LEN_META)
+
+            # merge existing assets with new-found, and create a list without duplicates
+            file_asset_ids = file.asset_ids or []
+            asset_ids_list = list(set(file_asset_ids + entities_id_found))
+
+            # If list of assets more than 1000 items, cut the list at 1000
+            if len(asset_ids_list) > 1000:
+                print(
+                    f"[WARNING] List of assetsIds for file {file.external_id} > 1000 ({len(asset_ids_list)}), "
+                    "cutting list at 1000 items"
+                )
+                asset_ids_list = asset_ids_list[:1000]
+
+            if config.debug:
+                print(f"[INFO] Converted and created (not upload due to DEBUG) file: {file_xid}")
+                print(f"[INFO] Assets found: {asset_names}")
+                continue
+
+            annotated_count += 1
+            # Note: add a minute to make sure annotation time is larger than last update time:
+            timestamp = datetime.now(timezone.utc) + timedelta(minutes=1).strftime(ISO_8601)
```

Comment:
```suggestion
            timestamp = datetime.now(timezone.utc) + timedelta(minutes=1)
```
you cannot add a string to a datetime

Reply:
Haha wtf was I thinking here 😂

Reply:
Ah, just forgot a pair of `()`